kedro/extras/datasets/spark/spark_dataset.py

```diff
@@ -309,9 +318,34 @@ def __init__(  # pylint: disable=too-many-arguments
         if save_args is not None:
             self._save_args.update(save_args)
 
+        # Handle schema load argument
+        self._schema = self._load_args.pop("schema", None)
+        if self._schema is not None:
+            if isinstance(self._schema, dict):
+                self._schema = self._load_schema_from_file(self._schema)
+
         self._file_format = file_format
         self._fs_prefix = fs_prefix
 
+    @staticmethod
+    def _load_schema_from_file(schema: Dict[str, Any]) -> StructType:
+        credentials = deepcopy(schema.get("credentials")) or {}
+        protocol, schema_path = get_protocol_and_path(schema["filepath"])
```

Comment:
Open Q: Do we wanna handle the missing `filepath` argument by a `DatasetError`?

Reply:
Maybe verify if the downstream exception raised by everything else is meaningful or not?

Reply:
Easiest answer is to add a test for this (schema with missing `filepath`) and see what it looks like. I suspect a `DataSetError` would be easier to read than the `KeyError` but worth checking as @datajoely said.

Reply:
Added test, covered it with a custom `SparkDataset` error.